### ISSUE TITLE
Include terminal name in terminal textarea title

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#99072]
Currently focusing the terminal reads something like "Terminal 1", "Terminal 2". It should read "Terminal 1, bash", "Terminal 2, (custom title)", etc.

Context: https://github.com/microsoft/vscode/issues/95573#issuecomment-637445449

**Comment by chrisnorman7** [comment]
Or simply the title you set... Maybe the title, then "Terminal 1" in brackets?

**Comment by chrisnorman7** [comment]
Or maybe go absolutely wild, and have it as a configurable string. Something like "$n (Terminal $i)", where $n is the name you set, and $i is the index.

**Comment by tabaddor** [comment]
I can look into fixing this issue if it is still open

**Comment by Tyriar** [comment]
@tabaddor please do 🙂, the backlog for us essentially means we want to do this eventually.

**Comment by tabaddor** [comment]
On it. @chrisnorman7 Quick clarification, but what was meant by "custom title" in the issue description above? 

I guess more specifically, with `Terminal X, (custom)`, is the (custom) the user's choice? How will that terminal name be determined?

**Comment by Tyriar** [comment]
@tabaddor by custom I mean when you rename a terminal via this command:

![image](https://user-images.githubusercontent.com/2193314/84062167-d9e1ab80-a973-11ea-9c94-1bc0b751d552.png)

**Comment by chrisnorman7** [comment]
Not sure what the image says, but when you rename the terminal with the
"Terminal: Rename" command.

Thanks so much for dealing with it.

Take care,

Chris Norman

**Comment by tabaddor** [comment]
I anticipate changing this line:

> https://github.com/microsoft/vscode/blob/162abc77c787ff7a31ad221277dc7c9d0b437ade/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L487

to this: 

`xterm.textarea.setAttribute('aria-label', nls.localize('terminalTextBoxAriaLabel', "Terminal {this._id}, {this._title}"));`

Though prior to setting this attribute I anticipate I will need to add a simple check to see if the title was set or not, along with ensuring this attribute is updated on title changes (renamed instances). Although the latter should already be accounted for.

**Comment by Tyriar** [comment]
To get localization to work it will need to be `"Terminal {0}, {1}", this._id, this._title)`, and yes good thinking we should have an if in there are 2 separate `nls.localize` calls, one with and one without the title.

**Comment by tabaddor** [comment]
Will a submit a PR, but any way to get around this:

![error](https://user-images.githubusercontent.com/44756122/84573686-18aaa380-ad70-11ea-9040-93e277e75250.png)

Was thinking of using a ternary conditional operator, but would see the same warning.

**Comment by roblourens** [comment]
Where should I see this? When I focus the terminal, voiceover reads "Terminal 1, complentary, group"

**Comment by alexr00** [comment]
I'm also not getting what I'd expect from this issue. 

What's read:
Terminal 2  edit
blank

Red dot where I clicked:
![image](https://user-images.githubusercontent.com/38270282/86607146-30f59300-bfa9-11ea-83fe-5a14c8f2a72c.png)

What I'd expect that to read: "Terminal 2 powershell"

**Comment by sunwei-ms** [comment]
I'm going to take a shot at this for Grace Hopper's Open Source Day!

**Comment by plainerman** [comment]
I see that this issue hasn't been fully resolved as of now. I would like to try to fix it if it is still available to work on. 

As I am completely new to this project, I would appreciate some code pointers. With the pointers above I have already found out where the string "said by the narrator" is set. But could someone point me to the entry point for `Terminal:Rename`?

Thank you, looking forward to making my first contribution.

**Comment by plainerman** [comment]
I found the entry point of `Terminal:Rename` (if anyone is searching for it, see [here](https://github.com/microsoft/vscode/blob/b2444f539bc5e3bbc099ea6ee453e0dd7b3f4385/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L952)) and have resolved the issue.

A pull request has been created at #112317.

**Comment by isidorn** [comment]
Thanks a lot for creating a PR!
